### ISSUE TITLE
[camel-main] Wiremocked tests do not work after upgrade of jetty

### ISSUE THREAD

**Issue by JiriOndrusek** [apache/camel-quarkus#5592]
### Bug description

Camel brought an update of jetty which caiuses following error (for each integration test using wiremock)

```
2023-12-06T08:58:43.7966773Z Caused by: java.lang.IncompatibleClassChangeError: class org.eclipse.jetty.http2.server.HttpChannelOverHTTP2 has interface org.eclipse.jetty.server.HttpChannel as super class
2023-12-06T08:58:43.7968686Z 	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
2023-12-06T08:58:43.7969824Z 	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1017)
2023-12-06T08:58:43.7971156Z 	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:508)
2023-12-06T08:58:43.7972719Z 	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:468)
2023-12-06T08:58:43.7974570Z 	at org.eclipse.jetty.http2.server.AbstractHTTP2ServerConnectionFactory.<init>(AbstractHTTP2ServerConnectionFactory.java:80)
2023-12-06T08:58:43.7976507Z 	at org.eclipse.jetty.http2.server.HTTP2ServerConnectionFactory.<init>(HTTP2ServerConnectionFactory.java:53)
2023-12-06T08:58:43.7978293Z 	at org.eclipse.jetty.http2.server.HTTP2CServerConnectionFactory.<init>(HTTP2CServerConnectionFactory.java:53)
2023-12-06T08:58:43.7980096Z 	at org.eclipse.jetty.http2.server.HTTP2CServerConnectionFactory.<init>(HTTP2CServerConnectionFactory.java:48)
2023-12-06T08:58:43.7981934Z 	at com.github.tomakehurst.wiremock.jetty11.Jetty11HttpServer.createHttpConnector(Jetty11HttpServer.java:53)
2023-12-06T08:58:43.7983866Z 	at com.github.tomakehurst.wiremock.jetty.JettyHttpServer.<init>(JettyHttpServer.java:89)
2023-12-06T08:58:43.7985366Z 	at com.github.tomakehurst.wiremock.jetty11.Jetty11HttpServer.<init>(Jetty11HttpServer.java:44)
2023-12-06T08:58:43.7987100Z 	at com.github.tomakehurst.wiremock.jetty.JettyHttpServerFactory.buildHttpServer(JettyHttpServerFactory.java:31)
2023-12-06T08:58:43.7988903Z 	at com.github.tomakehurst.wiremock.WireMockServer.<init>(WireMockServer.java:75)
```

**Comment by jamesnetherton** [comment]
Not sure how best to handle this. WireMock is tied to Jetty 11.x. There's a couple of open issues for upgrading Jetty:

https://github.com/wiremock/wiremock/issues/2513
https://github.com/wiremock/wiremock/issues/2395

I also noticed this thing has been added to the BOM, which is probably wrong and needs investigating:

https://github.com/apache/camel-quarkus/blob/camel-main/poms/bom/src/main/generated/flattened-reduced-pom.xml#L7728

**Comment by jamesnetherton** [comment]
This is fixed by #5593 & https://github.com/apache/camel-quarkus/commit/0a7c928b805cd63a732e2f1e724bc42bf14743ec.
